### ISSUE TITLE
Implementation of erased_serde::Serialize for custom types

### ISSUE THREAD

**Issue by VladYermakov** [dtolnay/erased-serde#17]
How to implement `erased_serde::Serialize` manually for custom type?
The types `Ok` and `Error` are private and this prevents to implement `erased_serde::Serialize` for any type or trait outside the library. But if we can't implement `serde::Serialize` we can't have `erased_serde::Serialize` for this type.
E.g. for some trait `TheTrait` we need type-erased serialize because if we implement `serde::Serialize` we'll have `The trait TheTrait cannot be made into an object`, but we can't implement `erased_serde::Serialize` manually because `Ok` and `Error` of `Result` are private.
Please provide any solution or make `Ok` and `Error` types public.

**Comment by dtolnay** [comment]
Correct, you are not allowed to implement erased_serde::Serialize. There is an `impl<T: ?Sized> erased_serde::Serialize for T where T: serde::Serialize` so you need to provide a serde::Serialize impl and receive an automatic erased_serde::Serialize impl. The [`serialize_trait_object!`](https://docs.rs/erased-serde/0.3/erased_serde/macro.serialize_trait_object.html) macro implements serde::Serialize for any trait object.
